src/ert/_c_wrappers/enkf/ensemble_config.py

```diff
@@ -307,7 +317,7 @@ def _get_schedule_file_node(
             if isinstance(schedule_file, str):
                 file_path = schedule_file
             file_name = _get_filename(file_path)
-            options = _option_dict(file_path, 1)
+            options = _option_dict(schedule_file, 1)
```

Comment:
and of course i see before i had doubt about it being a list - this transformation from config content to dict is just impossible to understand (for me). i'm sorry... maybe we can look at it together, and you could just explain it to me...

Reply:
The transformation from config content to dict is not very easy to understand I only know that we will fix it once we get to the parsing refactoring. 

Also, I am keeping the check for schedule_file being a simple string because that is what was happening when ResConfig was being initialized using a "config" the infamous third initialization option

Reply:
ya OK cool, then it all makes sense! do you mind putting an actual comment in the code, saying this is only needed for everest? i've done this before, too, so you'll be in company. don't know if it's good company, but it's definitely company : D

Reply:
> so I still have my doubts about schedule file ever being a simple string - the config dict path creates a dict (implementation changes pending), and the config content path creates a list of lists of strings. is everest causing you to keep this in? or just caution, as it was in the original code (I'm asking, i didn't check)?

I took another more patient look at this schedule file thing and it seems I was working under the wrong assumption that this could be a list of schedule files and it turns out that is not the case, also removed the `str` guard it seems it is not needed anymore

Reply:
ok cool : )